RecoMuon/L2MuonSeedGenerator/src/L2MuonSeedGeneratorFromL1T.cc

```diff
@@ -124,246 +157,804 @@ void L2MuonSeedGeneratorFromL1T::produce(edm::Event& iEvent, const edm::EventSet
+
+                LogDebug(metname) << "L1 info: Det and State:";
+                LogDebug(metname) << debug.dumpMuonId(newTSOSDet->geographicalId());
+
+                if (newTSOS.isValid()){
+
+                  //LogDebug(metname) << "(x, y, z) = (" << newTSOS.globalPosition().x() << ", "
+                  //                  << newTSOS.globalPosition().y() << ", " << newTSOS.globalPosition().z() << ")";
+                  LogDebug(metname) << "pos: (r=" << newTSOS.globalPosition().mag() << ", phi="
+                                    << newTSOS.globalPosition().phi() << ", eta=" << newTSOS.globalPosition().eta() << ")";
+                  LogDebug(metname) << "mom: (q*pt=" << newTSOS.charge()*newTSOS.globalMomentum().perp() << ", phi="
+                                    << newTSOS.globalMomentum().phi() << ", eta=" << newTSOS.globalMomentum().eta() << ")";
+
+                  //LogDebug(metname) << "State on it";
+                  //LogDebug(metname) << debug.dumpTSOS(newTSOS);
+
+                  bool isAssoOffseed = isAssociateOfflineSeedToL1(offlineSeedHandle, dRmtx, newTSOS, imu, selOffseeds, dRcone );
+
+                  if(isAssoOffseed) {
+                    isAsso = true;
+                  }
+
+                  // Using old way
+                  else {
+                    if(useUnassociatedL1) {
+                      // convert the TSOS into a PTSOD
+                      PTrajectoryStateOnDet const & seedTSOS = trajectoryStateTransform::persistentState( newTSOS,newTSOSDet->geographicalId().rawId());
+                      output->push_back(L2MuonTrajectorySeed(seedTSOS,container,alongMomentum,
+                                        MuonRef(muColl,  distance(muColl->begin(muColl->getFirstBX()),it)  )));
+                    }
+                  }
+
+                }
+              }
+            }
+          } // useOfflineSeed
+
+          else {
+            // convert the TSOS into a PTSOD
+            PTrajectoryStateOnDet const & seedTSOS = trajectoryStateTransform::persistentState( tsos, theid.rawId());
+            output->push_back(L2MuonTrajectorySeed(seedTSOS,container,alongMomentum,
+                              MuonRef(muColl,  distance(muColl->begin(muColl->getFirstBX()),it)  )));
+          }
+
         }
-        else {
-          // convert the TSOS into a PTSOD
-          PTrajectoryStateOnDet const & seedTSOS = trajectoryStateTransform::persistentState( tsos, theid.rawId());
-          output->push_back(L2MuonTrajectorySeed(seedTSOS,container,alongMomentum,
-                            MuonRef(muColl,  distance(muColl->begin(muColl->getFirstBX()),it)  )));
-        }      
-      }  
+      }
+    }
+
+    // MatchType : 0 Old matching,   1 L1 Order(1to1), 2 Min dR(1to1), 3 Higher Q(1to1), 4 All matched L1
+    if(useOfflineSeed && isAsso) {
+      unsigned int NofflineSeed1 = offlineSeedHandle->size();
+
+      unsigned int nL1;
+      unsigned int i, j; // for the matrix element
+
+      /*
```

Comment:
Please remove the commented out code that you included for your debug (or use some LOgDebug if you want to maintain here for possible future needs)